### DISCUSSION TITLE
Simpler way to run steps only for a single platform

### DISCUSSION THREAD

**Discussion by DanTup** [community/community#25894] [Actions]
<p>I have some conditions in my steps like:</p>
<pre><code>if: matrix.os == 'ubuntu-latest'
</code></pre>
<p>It feels clunky having to repeat “ubuntu-latest” in many places when really what I want to say is just “linux”. If I was running on multiple versions of Ubuntu or in the future if there were non-Ubuntu linux options, it would be weird to have to maintain all of these.</p>
<p>Is there some way we could have a field that represents the platform (“win”, “mac”, “linux” or similar)?</p>
<p>(I just found some people doing “startsWith(matrix.name, ‘linux’)” which looks slightly better - but still weird, especially as I think it needs to Ubuntu, not Linux).</p>

**Comment by wipe2238** [comment, marked as answer]
<pre><code>if: runner.os == 'Linux'

if: runner.os == 'Windows'

if: runner.os == 'macOS'
</code></pre>

**Comment by letmaik** [comment]
<p>Looks like <code>runner.os</code> is not documented, I’m wondering if this should be the official way to do it.</p>

**Comment by ethomson** [comment]
<p>We haven’t finalized the documentation for the <code>runner</code> context yet, but you should be able to safely rely on <code>runner.os</code>.</p>
